software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractControllerImpl.java

```diff
@@ -317,6 +331,8 @@ protected void computePortsAndUrls() {
 
         sensors().set(PROTOCOL, inferProtocol());
         sensors().set(MAIN_URI, URI.create(inferUrl()));
+        sensors().set(MAIN_URI_MAPPED_SUBNET, URI.create("http://" + sensors().get(SUBNET_ADDRESS) + ":" + sensors().get(PROXY_HTTP_PORT) + "/"));
+        sensors().set(MAIN_URI_MAPPED_PUBLIC, URI.create(inferUrlPublic()));
```

Comment:
@neykov Interesting approach, I didn't know about those enrichers. I did a quick test and the  `OnSubnetNetworkEnricher` works whereas `OnPublicNetworkEnricher` does as @aledsage explained.

Although, using enrichers would probably be a cleaner solution, we can go back to it later.